Dockerfile

```diff
@@ -4,10 +4,12 @@ FROM python:3.9-slim-buster
 
 WORKDIR /app
 
-# Note: Eventaully many files will be omitted. Images build with current Dockerfile should not be pushed to a public repository.
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt 
```

Comment:
Why not move this command after the `COPY . .` on line 11 below, and save the explicit COPY of `requriments.txt` above on line 7

Reply:
@Andrew-Dickinson  it builds faster this way.  Otherwise it always rebuilds the pip requirements.

Reply:
Mostly when building locally.

Reply:
Oh right